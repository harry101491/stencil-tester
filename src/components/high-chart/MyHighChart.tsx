import { Component, h, Element } from "@stencil/core";
import HighCharts from 'highcharts';

@Component({
  shadow: true,
  styleUrl: './MyHighChart.scss',
  tag: 'my-high-chart'
})
export class MyHighChart {

  @Element() el: HTMLElement;

  chart: any;

  options: HighCharts.Options = {
    title: {
      text: 'My Chart Title',
    },
    chart: {
      type: 'spline',
    },
    subtitle: {
      text: 'My Chart Subtitle',
    },
    tooltip: {
      backgroundColor: '#FCFFC5',
      borderColor: 'black',
      borderRadius: 10,
      borderWidth: 3
    },
    series: [
      {
        name: 'first-series',
        data: [1, 2, 1, 4, 3, 6],
        type: undefined,
      }
    ]
  };

  render() {
    return (
      <div
        id='container'
        style={{ height: '400px', width: '100%' }}
      >

      </div>
    );
  }

  componentDidLoad() {
    this.chart = this.el.shadowRoot.querySelector('#container');
    new HighCharts.Chart(this.chart, this.options);
  }
}
