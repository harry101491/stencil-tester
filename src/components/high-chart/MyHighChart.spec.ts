import { MyHighChart } from './MyHighChart';
// import { newSpecPage } from '@stencil/core/testing';

describe('My High Chart Test Suite', () => {
  let currentComponent: MyHighChart;

  beforeAll(async () => {
    currentComponent = new MyHighChart();
    currentComponent.options = {
      chart: {
        type: 'test'
      },
      title: {
        text: 'My Test chart'
      },
      series: [
        {
          name: 'first-series',
          data: [1, 2, 1, 4, 3, 6],
          type: undefined,
        }
      ]
    };
  });

  it('Component should render', () => {
    expect(currentComponent).toMatchSnapshot();
  });

  // it('componentDidLoad() must instantiate HighChart Chart class', async () => {
  //   const page = await newSpecPage({
  //     components: [MyHighChart],
  //     html: `<my-high-chart></my-high-chart>`
  //   });
  //   const highChart = page.body.querySelector('my-high-chart');
  //   expect(highChart).not.toBeNull();
  // });
});

// currentComponent.el = {
//   shadowRoot: {
//     querySelector: jest.fn().mockImplementation(() => {}),
//     ...currentComponent.el.shadowRoot,
//   },
//   ...currentComponent.el,
// };
// currentComponent.componentDidLoad();
// expect(HighCharts.Chart).toHaveBeenCalledWith(currentComponent.chart, currentComponent.options);
