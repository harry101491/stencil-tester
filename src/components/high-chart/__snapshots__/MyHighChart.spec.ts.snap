// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My High Chart Test Suite Component should render 1`] = `
MyHighChart {
  "options": Object {
    "chart": Object {
      "type": "test",
    },
    "series": Array [
      Object {
        "data": Array [
          1,
          2,
          1,
          4,
          3,
          6,
        ],
        "name": "first-series",
        "type": undefined,
      },
    ],
    "title": Object {
      "text": "My Test chart",
    },
  },
}
`;
